src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java

```diff
@@ -0,0 +1,111 @@
+import com.pubnub.api.models.consumer.access_manager.v3.PNRevokeTokenResult;
+import com.pubnub.api.models.server.access_manager.v3.RevokeTokenResponse;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import retrofit2.Call;
+import retrofit2.Response;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Accessors(chain = true, fluent = true)
+public class RevokeToken extends Endpoint<RevokeTokenResponse, PNRevokeTokenResult> {
+
+    @Setter
+    private String token;
+
+    public RevokeToken(PubNub pubnub,
+                       TelemetryManager telemetryManager,
+                       RetrofitManager retrofit,
+                       TokenManager tokenManager) {
+        super(pubnub, telemetryManager, retrofit, tokenManager);
+    }
+
+    @Override
+    protected List<String> getAffectedChannels() {
+        return null;
+    }
+
+    @Override
+    protected List<String> getAffectedChannelGroups() {
+        return null;
+    }
+
+    @Override
+    protected void validateParams() throws PubNubException {
+        if (this.getPubnub().getConfiguration().getSecretKey() == null || this.getPubnub()
+                .getConfiguration()
+                .getSecretKey()
+                .isEmpty()) {
+            throw PubNubException.builder().pubnubError(PubNubErrorBuilder.PNERROBJ_SECRET_KEY_MISSING).build();
+        }
+        if (this.getPubnub().getConfiguration().getSubscribeKey() == null || this.getPubnub()
+                .getConfiguration()
+                .getSubscribeKey()
+                .isEmpty()) {
+            throw PubNubException.builder().pubnubError(PubNubErrorBuilder.PNERROBJ_SUBSCRIBE_KEY_MISSING).build();
+        }
+        if (this.token == null) {
+            throw PubNubException.builder()
+                    .pubnubError(PubNubErrorBuilder.PNERROBJ_TOKEN_MISSING)
+                    .build();
+        }
+
+    }
+
+    @Override
+    protected Call<RevokeTokenResponse> doWork(Map<String, String> baseParams) throws PubNubException {
+        return getRetrofit()
+                .getAccessManagerService()
+                .revokeToken(getPubnub().getConfiguration().getSubscribeKey(), repairEncoding(token), baseParams);
+    }
+
+    @Override
+    protected PNRevokeTokenResult createResponse(Response<RevokeTokenResponse> input) throws PubNubException {
+        if (input.body() == null) {
+            return null;
+        }
+
+        return new PNRevokeTokenResult();
+    }
+
+    @Override
+    protected PNOperationType getOperationType() {
+        return PNOperationType.PNAccessManagerRevokeToken;
+    }
+
+    @Override
+    protected boolean isAuthRequired() {
+        return false;
+    }
+
+    private String repairEncoding(String token) {
+        String[] parts = token.split(" ");
```

Comment:
We never had a case where token comes with spaces. Any reason for splitting token with space?

Reply:
Yes. A test scenario in contract tests. That's why I disagree with the approach. I had to jump over some hoops just to satisfy the test

Reply:
https://github.com/pubnub/sdk-specifications/blob/62e194d895dfac0e970dc2f6d9f3f76815de5b92/features/access/revoke-token.feature#L43

Reply:
Hm..I am surprised. I thought existing SDK has already logic of encoding chars. 
For example; When we get space in channel names or UUID, we encode it. So this logic already exist. Why not reuse the same logic? Is SDK encoding separately for each operation?

Reply:
Well, we have it working for channels and most other parameters but it's done by okhttp which do not encodes everything that we have in the test case. Specifically it's not encoding the `=`. If I encode it manually and pass it to okhttp with `%20` okhttp will encode the `%` once again. So I had to disable the encoding for `token` done by okhttp and then encode it myself. Unfortunately the encoder used encodes `space` into `+` and not `%20`